### PR TITLE
Only use one single ciphersuite if 0-rtt is actually enabled

### PR DESCRIPTION
**Pull request by zhihan**
In the original code, we break out of the loop if the compiler flag is set. It reads wrong to me. I think we should limit the behavior to where 0-RTT is enabled in the configuration and the handshake is set up for resumption.

### CODE REVIEW

**Comment by hannestschofenig** [comment]
Looks good to me.

**Review by hanno-becker — CHANGES_REQUESTED** [review]
Thank you @zhihan, it looks good to me. However, could you add a test to `ssl-opt.sh` to check the behaviour? It should be easy by grepping for two distinct ciphersuites in the output.

File: tests/ssl-opt.sh [hunk 1]
```diff
@@ -1499,7 +1499,7 @@ run_test    "TLS 1.3, TLS_AES_256_GCM_SHA384, ext PSK, early data" \
             "$P_SRV nbio=2 debug_level=5 force_version=tls1_3 early_data=-1 key_exchange_modes=psk psk=010203 psk_identity=0a0b0c" \
             "$P_CLI nbio=2 debug_level=5 force_version=tls1_3 force_ciphersuite=TLS_AES_256_GCM_SHA384 key_exchange_modes=psk early_data=1 psk=010203 psk_identity=0a0b0c" \
             0 \
-	    -s "found early_data extension"                 \
+        -s "found early_data extension"                 \
```
Comment (zhihan) [inline tests/ssl-opt.sh:1502]: Sorry, this was added by my editor: \tab becomes four spaces.

File: library/ssl_tls13_client.c [hunk 2]
```diff
@@ -1752,9 +1752,10 @@ static int ssl_client_hello_write_partial( mbedtls_ssl_context* ssl,
 
         buflen -= 2;
 
-#if defined(MBEDTLS_ZERO_RTT)
-        /* For ZeroRTT we only add a single ciphersuite. */
-        break;
+#if defined(MBEDTLS_ZERO_RTT) && defined(MBEDTLS_SSL_NEW_SESSION_TICKET)
+        /* For 0-RTT we only add a single ciphersuite. */
+        if( ssl->conf->early_data_enabled == MBEDTLS_SSL_EARLY_DATA_ENABLED)
+            break;
```
Comment (zhihan) [inline library/ssl_tls13_client.c:1758]: Remove the check for resumption, this condition should be guarded by early_data alone.

**Comment by zhihan** [comment]
> Thank you @zhihan, it looks good to me. However, could you add a test to `ssl-opt.sh` to check the behaviour? It should be easy by grepping for two distinct ciphersuites in the output.

Added tests.

**Review by hanno-becker — APPROVED** [review]
LGTM

**Pull request merged**